crates/turbopack-core/src/chunk/mod.rs

```diff
@@ -444,13 +450,14 @@ async fn chunk_content_internal<I: FromChunkableAsset>(
                 // not loaded in parallel
                 let mut inner_chunk_groups = Vec::new();
 
-                for asset in assets
-                    .await?
-                    .iter()
-                    .filter(|asset| processed_assets.insert((chunking_type, **asset)))
-                {
-                    let asset: &AssetVc = asset;
-
+                for asset in result.await?.primary.iter().filter_map(|result| {
```

Comment:
Nit: Can we assign this RHS to a variable. The `for` nesting here becomes weird because of the multi-line function.